components/browser/engine-system/src/test/java/mozilla/components/browser/engine/system/SystemEngineViewTest.kt

```diff
@@ -230,6 +230,30 @@ class SystemEngineViewTest {
         assertEquals("https://www.firefox.com/", engineView.currentUrl)
     }
 
+    @Test
+    fun `WebView client notifies navigation state changes`() {
+        val engineSession = SystemEngineSession()
+
+        val engineView = SystemEngineView(RuntimeEnvironment.application)
+        val observer: EngineSession.Observer = mock()
+        val webView: WebView = mock()
+        `when`(webView.canGoBack()).thenReturn(true)
+        `when`(webView.canGoForward()).thenReturn(true)
+
+        // This sets the currentUrl.
+        engineView.currentWebView.webViewClient.onPageStarted(webView, "https://www.mozilla.org/", null)
+
+        // No observers notified when session isn't rendered.
+        engineSession.register(observer)
+        verify(observer, never()).onNavigationStateChange(true, true)
```

Comment:
Optional nit: this definitely works, but it might be a bit more robust as `onNavigationStateChange(any(), any())`